x/liquidity/keeper/swap.go

```diff
@@ -260,43 +260,37 @@ func (k Keeper) CancelOrder(ctx sdk.Context, msg *types.MsgCancelOrder) error {
 
 // CancelAllOrders handles types.MsgCancelAllOrders and cancels all orders.
 func (k Keeper) CancelAllOrders(ctx sdk.Context, msg *types.MsgCancelAllOrders) error {
-	var canceledOrderIds []string
-	cb := func(pair types.Pair, order types.Order) (stop bool, err error) {
-		if order.Orderer == msg.Orderer && order.Status != types.OrderStatusCanceled && order.BatchId < pair.CurrentBatchId {
-			if err := k.FinishOrder(ctx, order, types.OrderStatusCanceled); err != nil {
-				return false, err
-			}
-			canceledOrderIds = append(canceledOrderIds, strconv.FormatUint(order.Id, 10))
+	pairIdSet := map[uint64]struct{}{} // set of pairs where to cancel orders
+	var pairIds []string               // needed to emit an event
+	for _, pairId := range msg.PairIds {
+		if _, found := k.GetPair(ctx, pairId); !found { // check if the pair exists
+			return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "pair %d not found", pairId)
 		}
-		return false, nil
+		pairIdSet[pairId] = struct{}{} // add pair id to the set
+		pairIds = append(pairIds, strconv.FormatUint(pairId, 10))
```

Comment:
If cache `orderPairCache` here also, Wouldn't it be possible to increase the cache hit ratio?

Reply:
Thanks, applied the change.